### PR TITLE
CHECK-61

### PR DESCRIPTION
**Pull request by JayJayAquino**
# Description

Help Create TimeStamp object and added unit tests for said object

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

run phpunit to ensure tests pass(specifically TimeStampTest), load checkpoint to ensure its not broken

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/DomainValueObjects/Log/TimeStamp/TimeStamp.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+
+        $this->uuid = $uuid;
+        $this->timeStamp = $timeStamp;
+        $this->validation();
+    }
+
+    private function validation()
+    {
+        if($this->timeStamp == null || $this->timeStamp == ''){
```
Comment (ChanM0) [inline app/DomainValueObjects/Log/TimeStamp/TimeStamp.php]: can you make these one line?

File: app/Http/Controllers/Api/Log/ClockOutDomain/Services/ClockOutService.php [hunk 2]
```diff
@@ -58,15 +58,13 @@ public function clockOut(string $currentLocation, string $timeStamp, string $log
         
         $userLocation = $this->timePuncherRetriever->getUserLocation($user, $currentLocation);
 
-        $clockOut = $this->getLogParam($userLocation, $timeStamp);
-         
+        $clockOut = $this->getLogParam($userLocation, $timeStamp);
```
Comment (ChanM0) [inline app/Http/Controllers/Api/Log/ClockOutDomain/Services/ClockOutService.php:61]: can you include a space after clockout.

**Review by ChanM0 — APPROVED** [review]
👍

**Review by digitalsoba — APPROVED** [review]
👍

**Pull request merged**